crates/trie/db/src/storage.rs

```diff
@@ -82,16 +86,28 @@ impl<'a, TX: DbTx> DatabaseStorageRoot<'a, TX>
 impl<TX: DbTx> DatabaseHashedStorage<TX> for HashedStorage {
     fn from_reverts(tx: &TX, address: Address, from: BlockNumber) -> Result<Self, DatabaseError> {
         let mut storage = Self::new(false);
-        let mut storage_changesets_cursor = tx.cursor_read::<tables::StorageChangeSets>()?;
-        for entry in storage_changesets_cursor.walk_range(BlockNumberAddress((from, address))..)? {
-            let (BlockNumberAddress((_, storage_address)), storage_change) = entry?;
-            if storage_address == address {
-                let hashed_slot = keccak256(storage_change.key);
-                if let hash_map::Entry::Vacant(entry) = storage.storage.entry(hashed_slot) {
-                    entry.insert(storage_change.value);
-                }
-            }
+        for (slot, value) in from_reverts(tx, address, from)? {
+            storage.storage.insert(keccak256(slot), value);
         }
         Ok(storage)
     }
 }
+
+/// Retrieves the original storage values from changed slots since the specified block and address.
+pub fn from_reverts<TX: DbTx>(
```

Comment:
Basically extracted from the code above, since `HashedStorage` was only used for trie updating logic where we want "hashed" storage since keys in the tree are hashes.

But for flatdb, we're interested in plain addresses and storage slots and not hashes. So basically we use the same logic for both now.